### PR TITLE
feat: filtering for themes

### PR DESCRIPTION
**Pull request by yu-melody**
# Description

[Link to Ticket](https://github.com/GenerateNU/specialstandard/issues/32)

This PR adds comprehensive filtering capabilities to the GET /themes endpoint, allowing users to filter themes by month, year, and name search. All filters are optional and can be combined with existing pagination functionality.

## Features Implemented
Query Parameter Filters
- month - Filter themes by month (1-12)
    - Example: GET /themes?month=3 (March themes)
- year - Filter themes by year (1900-2100)
    - Example: GET /themes?year=2024 (2024 themes)
- search - Fuzzy search by theme name (case-insensitive, partial matching)
    - Example: GET /themes?search=spring (themes containing "spring")
## Combined Filtering
- All filters can be used together
    - Example: GET /themes?month=3&year=2024&search=activities
- Filters work seamlessly with existing pagination
    - Example: GET /themes?month=3&page=2&limit=5

## Input Validation
- Month: Must be 1-12 (January-December)
- Year: Must be 1900-2100
- Search: Maximum 255 characters
- Invalid parameters return 400 Bad Request with specific error messages

## Search Features
- Case-insensitive: spring, SPRING, Spring all work
- Partial matching: act matches "Spring Activities"
- Substring search: 2024 matches any theme containing "2024"
- PostgreSQL ILIKE: Robust database-level fuzzy search

## Backward Compatibility
- Existing API calls continue to work unchanged
- No breaking changes to response format
- Default behavior (no filters) returns all themes

# How Has This Been Tested?
Implemented individual and combined filter unit tests, pagination and filters integration, and validation edge cases.

Filtering for just 2025:
Screenshot 2025-09-28 at 1 23 54 PM

Filtering for 2024, May
Screenshot 2025-09-28 at 1 24 39 PM

Filtering for string that's something like "de may"
Screenshot 2025-09-28 at 1 25 25 PM

# Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have written unit tests for my code and tested it manually
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated the OpenAPI spec, if needed

### CODE REVIEW

File: backend/internal/service/handler/theme/handler_test.go [hunk 1]
```diff
@@ -224,7 +224,195 @@ func TestHandler_GetThemes(t *testing.T) {
...
+		{
+			name:           "Invalid month filter - negative",
+			url:            "?month=-1",
+			mockSetup:      func(m *mocks.MockThemeRepository) {},
+			expectedStatus: fiber.StatusBadRequest,
+			wantErr:        true,
+		},
+		{
+			name:           "Invalid year filter - too high",
+			url:            "?year=2101",
```
Comment (yu-melody) [inline backend/internal/service/handler/theme/handler_test.go:353]: I forget if we had agreed to place a validation on the year, and what that limit was?

Comment (michelleli04) [reply backend/internal/service/handler/theme/handler_test.go:353]: Nah we didn't but what you have is fine

File: backend/internal/storage/postgres/schema/themes.go [hunk 2]
```diff
@@ -43,14 +43,46 @@ func (r *ThemeRepository) CreateTheme(ctx context.Context, input *models.CreateT
...
 
-	rows, err := r.db.Query(ctx, query, pagination.Limit, pagination.GettOffset())
+	if len(conditions) > 0 {
+		query += " WHERE " + strings.Join(conditions, " AND ")
+	}
+
+	query += fmt.Sprintf(` ORDER BY year DESC, month DESC
+	LIMIT $%d OFFSET $%d`, argCount, argCount+1)
+	
+	args = append(args, pagination.Limit, pagination.GettOffset())
```
Comment (michelleli04) [inline backend/internal/storage/postgres/schema/themes.go:83]: haha lol didn't catch this typo from before

**Review by michelleli04 — APPROVED** [review]
LGTM!
